### DISCUSSION TITLE
Migrating Milvus data without MySQL connectivity

### DISCUSSION THREAD

**Discussion by akashmanna** [milvus-io/milvus#4579] [Q&A and General discussion] [answered]
Hi, I have a Milvus (version 10.2) cluster running with using MySQL for Metadata Management
I want to migrate the entire data from this cluster to another cluster, but there is no connectivity to the existing MySQL from the new cluster. Furthermore, I don't have permissions to export data from the existing MySQL. 

Is there a way to migrate data without migrating the contents of the MySQL database?

**Comment by yhmo** [comment, marked as answer]
Meta must be consist with entity data. If the meta is stored in MySQL, milvus cannot work without MySQL.
If the entity data is few, let's say, only one collection, several segments, theoretically we can manually create a meta. Anyway, it is difficult and could make mistake.

**Reply by akashmanna** [reply]
Thanks for the reply!

Just being curious, so in case if the MySQL goes down or the table is accidentally deleted, does this mean there is no way to meaningfully restore the data?

**Reply by yhmo** [reply]
If MySQL is shutdown, most of api will failed and return error. If someone use mysql client to drop the table by sql such as "DROP TABLE TableFiles;", the milvus also can not work and throw unexpected error.

**Reply by akashmanna** [reply]
okay, thanks!

**Comment by akashmanna** [comment]
Hi, To migrate the contents of Milvus to a new cluster, the process is to copy the contents of existing ```db``` directory and Migrate the ```milvus``` table from MySQL to the newer Milvus ... Right?

**Reply by akashmanna** [reply]
and is there a way to migrate a particular collection only

**Reply by yhmo** [reply]
Milvus stores vectors data in `db` directory, and store meta in MySQL. 
To migrate contents to a new milvus server, firstly copy `db` directory to overwrite the new milvus's  `db` folder. And shutdown the old milvus server. Makesure the new milvus server is using same MySQL server.
Our technical-support team is developing an export tool, which can export a particular collection data from `db` directory.
